common/caller.go

```diff
@@ -0,0 +1,29 @@
+package common
+
+import (
+	"runtime"
+	"strings"
+)
+
+func callerInfo(i int) (string, string, int, bool) {
+	ptr, file, line, ok := runtime.Caller(i)
+	var fName string
+	if ok {
+		f := runtime.FuncForPC(ptr)
+		if f != nil {
+			// f.Name() is like: github.com/docker/libnetwork/common.MethodName
+			tmp := strings.Split(f.Name(), ".")
+			if len(tmp) > 0 {
+				fName = tmp[len(tmp)-1]
+			}
+		}
+	}
+
+	return fName, file, line, ok
+}
+
+// CallerName returns the name of the function at the specified level with additional information
+// level == 0 means current method name
+func CallerName(level int) (string, string, int, bool) {
```

Comment:
can this just return the function name. I don't think we will be using file and line number at all. We can use this directly in other debugs we have. I have closed the context PR docker/libnetwork#1835 because we can use this instead.